### PR TITLE
Prow: pass in list of trusted orgs as parameter to hook and store in github client

### PR DESCRIPTION
**Pull request by jfelten**
client.  Modify trigger plugin to use.

### CODE REVIEW

**Comment by spxtr** [comment]
@k8s-bot ok to test

This way of doing it is certainly possible, but I'm worried that we'll get too many confusing flags if we go this route for all config options. I started working on #2566 (issue #2546) but haven't given it much thought since, but the idea there is to add the trusted org to `config.yaml`.

What do you think?

**Comment by jfelten** [comment]
I'm fine with however you want to do it.  I agree that config would probably be cleaner than passing flags, and most of the code in this PR should be compatible with that.  Do you have a config structure in mind?

**Comment by spxtr** [comment]
I was going for this sort of thing:
```yaml
trigger:
- repos:
  - kubernetes
  - kubernetes-incubator
 ￼ - kubernetes-security
￼  - google/cadvisor
  trusted_org: kubernetes
- repos:
  - whoeverelse/whateverrepo
  trusted_org: whoeverelse
```

I can try to get #2566 in today, and then it should be natural to continue work from there. We'll need to add things like the gubernator URL as a template in the config, rather than hardcoded as they are now.

File: prow/cmd/hook/main.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ var (
 	githubBotName     = flag.String("github-bot-name", "", "Name of the GitHub bot.")
 	webhookSecretFile = flag.String("hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
 	githubTokenFile   = flag.String("github-token-file", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
+	trustedOrgs       = flag.String("trusted-orgs", "kubernetes", "comma deliminated list of trusted github organizations")
```
Comment (spxtr) [inline prow/cmd/hook/main.go]: For consistency, capitalize "comma" and end with period.

File: prow/cmd/hook/main.go [hunk 2]
```diff
@@ -75,6 +76,7 @@ func main() {
 		}
 		githubClient = github.NewFakeClient(*githubBotName)
 		githubClient.Logger = logrus.StandardLogger()
+		githubClient.SetTrustedOrgs(*trustedOrgs)
```
Comment (spxtr) [inline prow/cmd/hook/main.go]: Can this line be removed? Looks like we're calling it twice.

File: prow/github/client.go [hunk 3]
```diff
@@ -651,3 +652,41 @@ func (c *Client) FindIssues(query string) ([]Issue, error) {
 	}, &issSearchResult)
 	return issSearchResult.Issues, err
 }
+
+// set the trusted orgs
```
Comment (spxtr) [inline prow/github/client.go]: See effective go for doc comment conventions: https://golang.org/doc/effective_go.html#commentary

In particular, this should start with "SetTrustedOrgs" and be a complete sentence.

File: prow/github/client.go [hunk 4]
```diff
@@ -651,3 +652,41 @@ func (c *Client) FindIssues(query string) ([]Issue, error) {
...
+func (c *Client) SetTrustedOrgs(trustedOrgsList string) {
+	c.trustedOrgs = strings.Split(trustedOrgsList, ",")
+}
+
+// return true if an org is in the trusted list
```
Comment (spxtr) [inline prow/github/client.go]: This should start with "IsTrustedOrg" and be a complete sentence.

File: prow/github/client.go [hunk 5]
```diff
@@ -651,3 +652,41 @@ func (c *Client) FindIssues(query string) ([]Issue, error) {
...
+func (c *Client) IsTrustedOrg(org string) bool {
+
+	if c.trustedOrgs == nil {
```
Comment (spxtr) [inline prow/github/client.go]: This is unnecessary. You can range over `nil` just fine in go.

File: prow/github/client.go [hunk 6]
```diff
@@ -651,3 +652,41 @@ func (c *Client) FindIssues(query string) ([]Issue, error) {
...
+	}
+	for _, trustedOrg := range c.trustedOrgs {
+		if trustedOrg == org {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *Client) IsTrustedMember(user string) (bool, error) {
```
Comment (spxtr) [inline prow/github/client.go]: Add a doc comment.

File: prow/github/client.go [hunk 7]
```diff
@@ -651,3 +652,41 @@ func (c *Client) FindIssues(query string) ([]Issue, error) {
...
+		}
+		if trusted {
+			return true, nil
+		}
+
+	}
+	return false, nil
+}
+
+func (c *Client) GetTrustedOrgs() []string {
```
Comment (spxtr) [inline prow/github/client.go]: Just call this `TrustedOrgs()`.

File: prow/plugins/trigger/ic.go [hunk 8]
```diff
@@ -75,7 +76,7 @@ func handleIC(c client, ic github.IssueCommentEvent) error {
 			return err
 		}
 		if !trusted {
-			resp := fmt.Sprintf("you can't request testing unless you are a [%s](https://github.com/orgs/%s/people) member", trustedOrg, trustedOrg)
+			resp := fmt.Sprintf("you can't request testing unless you are a [%s](https://github.com/orgs/%s/people) member", strings.Join(c.GitHubClient.GetTrustedOrgs(), ","), strings.Join(c.GitHubClient.GetTrustedOrgs(), ","))
```
Comment (spxtr) [inline prow/plugins/trigger/ic.go]: This doesn't make sense if there are multiple orgs.

File: prow/plugins/trigger/pr.go [hunk 9]
```diff
@@ -83,7 +84,7 @@ I'm waiting for a [%s](https://github.com/orgs/%s/people) member to verify that
 %s
 </details>
 `
-	comment := fmt.Sprintf(commentTemplate, pr.User.Login, trustedOrg, trustedOrg, plugins.AboutThisBot)
+	comment := fmt.Sprintf(commentTemplate, pr.User.Login, strings.Join(trustedOrgs, ","), strings.Join(trustedOrgs, ","), plugins.AboutThisBot)
```
Comment (spxtr) [inline prow/plugins/trigger/pr.go]: This doesn't make sense if there are multiple trusted orgs.

File: prow/cmd/hook/main.go (hunk 2 above)
Comment (jfelten) [inline prow/cmd/hook/main.go]: That particular line is called when local is set and a fake github client is used.  The other one is for a standard run.  I'll consolidate both calls after the conditional block to simplify the code.

File: prow/cmd/hook/main.go (hunk 1 above)
Comment (jfelten) [inline prow/cmd/hook/main.go]: Will do.

File: prow/github/client.go (hunk 3 above)
Comment (jfelten) [inline prow/github/client.go]: will do.

File: prow/github/client.go (hunk 4 above)
Comment (jfelten) [inline prow/github/client.go]: cleaned up comments per go specification

File: prow/github/client.go (hunk 5 above)
Comment (jfelten) [inline prow/github/client.go]: ok - will remove.  NullPointerException is burned into my consciousness from Java.

File: prow/github/client.go (hunk 6 above)
Comment (jfelten) [inline prow/github/client.go]: will do.

File: prow/github/client.go (hunk 7 above)
Comment (jfelten) [inline prow/github/client.go]: ok will refactor

File: prow/plugins/trigger/ic.go (hunk 8 above)
Comment (jfelten) [inline prow/plugins/trigger/ic.go]: changed messaging to: "you can't request testing unless you are a member of one of these orgs: %s"

File: prow/plugins/trigger/pr.go (hunk 9 above)
Comment (jfelten) [inline prow/plugins/trigger/pr.go]: I changed the template to make more sense.  Let me know if you want different wording.

**Comment by jfelten** [comment]
@k8s-bot pull-test-infra-bazel test this

**Comment by kargakis** [comment]
+1 for a config option over a flag

**Comment by jfelten** [comment]
@kargakis The plan is to use flags until a config option is available.

File: prow/plugins/trigger/pr.go (hunk 9 above)
Comment (spxtr) [inline prow/plugins/trigger/pr.go]: The orgs shouldn't be in brackets, and it would be nice if they stay links.

**Comment by spxtr** [comment]
LG other than the nit in the comment string.

**Comment by kargakis** [comment]
@jfelten @spxtr https://github.com/kubernetes/test-infra/pull/2566 is already up, why not proceed with that approach since it seems that we have consensus in favor of it?

**Comment by jfelten** [comment]
I'll leave up to @spxtr to decide what approach he wants to use.  I can modify https://github.com/kubernetes/test-infra/pull/2943 to use the approach of 2566 if taht is the direction we want to go, and close this PR.

**Comment by jfelten** [comment]
closing to use alternative trusted org

**Pull request closed without merging**